### PR TITLE
unify f1 score

### PR DESCRIPTION
**Pull request by cuent**
## What does this PR do?

Refactor F1 score

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by cuent** [comment]
Should `F1` object be renamed to `F1Score`?

**Comment by stancld** [comment]
> Should [`F1`](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/classification/f_beta.py#L177) object be renamed to `F1Score`?

Yes, please use `F1Score` for the class name :]

File: docs/source/references/functional.rst [hunk 1]
```diff
@@ -122,7 +122,7 @@ dice_score [func]
 f1 [func]
 ~~~~~~~~~~~~~~~
```
Comment (stancld) [inline docs/source/references/functional.rst:123]:
```suggestion
f1_score [func]
~~~~~~~~~~~~~~~
```

**Review by Borda — COMMENTED** [review]
- [ ] deprecation warnings & back compatibility
- [ ] rename also `F1` class to be consistent (suggestion `F1Score`)
- [ ] add changelog

part of #729

**Review by Borda — CHANGES_REQUESTED** [review]
deprecation warnings & back-compatibility, see https://github.com/PyTorchLightning/metrics/pull/731#pullrequestreview-847145436

File: CHANGELOG.md [hunk 2]
```diff
@@ -77,6 +76,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * `SNR` -> `SignalNoiseRatio`
   * `SI_SNR` -> `ScaleInvariantSignalNoiseRatio`
 
+- Renamed `torchmetrics.functional.f1` to `torchmetrics.functional.f1_score` and `torchmetrics.F1` to `torchmetrics.F1Score` to unify name convention with other frameworks ([#731](https://github.com/PyTorchLightning/metrics/pull/731))
```
Comment (Borda) [inline CHANGELOG.md:79]:
```suggestion
- Renamed F-score metrics: ([#731](https://github.com/PyTorchLightning/metrics/pull/731))
  * `torchmetrics.functional.f1` -> `torchmetrics.functional.f1_score`
  * `torchmetrics.F1` -> `torchmetrics.F1Score`
```

**Review by Borda — APPROVED** [review]

File: docs/source/references/modules.rst [hunk 3]
```diff
@@ -280,10 +280,10 @@ ConfusionMatrix
 .. autoclass:: torchmetrics.ConfusionMatrix
     :noindex:
 
-F1
+F1Score
 ~~
```
Comment (stancld) [inline docs/source/references/modules.rst:284]:
```suggestion
F1Score
~~~~~~~
```

**Review by stancld — APPROVED** [review]

**Pull request merged**